File: Arrays&Hashing/347TopKFrequentElements.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        m = {}
        res = []
        values = [[] for i in range(len(nums)+1)]
        
        for num in nums:
            if num in m:
                m[num] = m[num] + 1
            else :
                m[num] = 1

        
        for n,c in m.items():
            values[c].append(n)

        for i in range(len(values)-1,0,-1):
            for val in values[i]:
                res.append(val)
                if len(res)==k :
                    return res
                
        

        

File: Arrays&Hashing/347TopKFrequentElements.cpp
#include<vector>
#include<unordered_map>
#include<algorithm>
class Solution {
public:
    std::vector<int> topKFrequent(std::vector<int>& nums, int k) {
        std::unordered_map<int,int> m;
        std::vector<std::pair<int,int>> pairs;

        for( int i = 0 ; i < nums.size() ; i++){
            m[nums[i]]++;
        }
        for(const auto& p : m){
            pairs.push_back(std::make_pair(p.second, p.first));
        }
        sort(pairs.begin(), pairs.end());
        std::vector<int> ret;
        for(auto p = pairs.rbegin(); p != pairs.rend(); p++){
            ret.emplace_back(p->second);
            k--;
            if(k==0) break;
        }
        return ret;
    }
};